ner_v2/detectors/numeral/number/standard_number_detector.py

```diff
@@ -65,104 +71,158 @@ def init_regex_and_parser(self, data_directory_path):
+                })
+                original_list.append(original_text)
         return number_list, original_list
 
-    def _detect_numeric_digit(self, number_list, original_list):
+    def _detect_number_from_digit(self, number_list=None, original_list=None):
         """
-        Detect number from numeric text
+        Detect number from numeric text like 200 , 2.2k , 300.12
         Args:
             number_list (list): list containing detected numeric text
             original_list (list): list containing original numeral text
         Returns:
             number_list (list): list containing updated detected numeric text
             original_list (list): list containing updated original numeral text
+
+        Examples:
+            [In]  >>  self.processed_text = "200"
+            [In]  >>  _detect_number_from_digit()
+            [Out] >> ([{'value': '200', 'unit': None}], ['200'])
+
+            [In]  >>  self.processed_text = "200-300"
+            [In]  >>  _detect_number_from_digit()
+            [Out] >> ([{'value': '200', 'unit': None}, {'value': '300', 'unit': None}],
+                      ['200', '300'])
+
+            [In]  >>  self.processed_text = "1 2 3"
+            [In]  >>  _detect_number_from_digit()
+            [Out] >> ([{'value': '2', 'unit': None}, {'value': '2', 'unit': None}, {'value': '3', 'unit': None}],
+                      ['1', '2', '3'])
+
+            [In]  >>  self.processed_text = "12.23"
+            [In]  >>  _detect_number_from_digit()
+            [Out] >> ([{'value': '12.23', 'unit': None}], ['12.23'])
+
+            [In]  >>  self.processed_text = "2k"
+            [In]  >>  _detect_number_from_digit()
+            [Out] >> ([{'value': '2000', 'unit': None}], ['2k'])
+
+            [In]  >>  self.processed_text = "2.2k"
+            [In]  >>  _detect_number_from_digit()
+            [Out] >> ([{'value': '2200', 'unit': None}], ['2.2k'])
+
         """
         number_list = number_list or []
         original_list = original_list or []
-        numbers_char_set = set(self.language_number_map.keys())
 
-        for word in self.processed_text.split():
-            word_chars = list(word)
-            word_chars = [ch for ch in word_chars if ch != ',']
-            if not (set(word_chars) - numbers_char_set):
-                number = "".join([str(self.language_number_map[ch]) for ch in word_chars])
-                if self.max_digit >= len(number) >= self.min_digit:
-                    number_list.append(number)
-                    original_list.append(word)
+        patterns = self.regex_numeric_patterns.findall(self.processed_text)
+        for pattern in patterns:
+            original_text = pattern[0].strip()
+            number = pattern[1].replace(",", "")
+            scale = pattern[2].strip()
+            scale = self.language_scale_map[scale] if scale else 1
+            number = float(number) * scale
+            number = int(number) if number.is_integer() else number
```

Comment:
https://www.quora.com/How-numbers-are-written-in-Tamil

Few answers here explain that digits are not widely used and if they are used they use a different system
`2785` is `௨௲௭௱௮௰௫` which computes like `two x thousand  + seven x hundred + eight x ten + five`

Reply:
```
In [22]: a = re.compile(r'(([\d,]+[.]?[\d]*)\s?(k|m)?)\s*', re.UNICODE)
In [23]: b = a.findall(u'௩௰௨')
In [24]: b
Out[24]: [(u'\u0be9', u'\u0be9', u''), (u'\u0be8', u'\u0be8', u'')]
```
Yes tamil case works like roman numbers, as seen from above, it detects only `௩` and `௨` not 
`௰` which corresponds to `10` which in tamil not made from combination of `1` and `0`.

We should decide this, because this cannot be handled from current logic.